tools/importPlanGoals.js

```diff
@@ -0,0 +1,171 @@
+  return itemId;
+}
+
+/**
+ * Processes data from .csv creating data arrays and then inserting it to the database
+ */
+async function importGoals() {
+  try {
+    const cleanRoles = [];
+    const cleanTopics = [];
+    const cleanRoleTopics = [];
+    const cleanGoals = [];
+    const cleanGrantees = [];
+    const cleanTtaPlans = [];
+    const cleanTopicGoals = [];
+
+    await grantees.forEach((el, inx) => {
+      let currentGrantee;
+      let currentGranteeId;
+      let grants;
+      let currentGrants = [];
+      let currentGoalName;
+      let currentGoal = {};
+      let currentGoalId;
+      const currentGoals = [];
+
+      Object.keys(el).forEach((key) => {
+        if (key && key.startsWith('Grantee (distinct')) {
+          currentGrantee = el[key] ? el[key].split('|')[0] : `Unknown Grantee ${inx}`;
+          currentGranteeId = getItemId(cleanGrantees, currentGrantee.trim());
+          grants = el[key] ? el[key].split('|')[1] : `Unknown Grant ${inx}, Unknown Grant ${inx + 10000}`;
+          currentGrants = grants.split(',');
+        } else if (key && key.startsWith('Goal')) {
+          const goalColumn = key.split(' ');
+          let column;
+          if (goalColumn.length > 2) {
+            column = goalColumn[2].toLowerCase();
+            if (column === 'topics') {
+              const allTopics = el[key].split('\n');
+              allTopics.forEach((topicPipeRoles) => {
+                const topic = topicPipeRoles.split('|')[0];
+                const roles = topicPipeRoles.split('|')[1];
+                const trimmedTopic = topic.trim();
+                if (trimmedTopic !== '') {
+                  const topicId = getItemId(cleanTopics, trimmedTopic);
+                  if (roles) {
+                    const rolesArr = roles.split(',');
+                    rolesArr.forEach((role) => {
+                      const trimmedRole = role.trim();
+                      const roleId = getItemId(cleanRoles, trimmedRole);
+                      // associate topic with roles
+                      if (!cleanRoleTopics.some((e) => e.roleId === roleId
+                                                      && e.topicId === topicId)) {
+                        cleanRoleTopics.push({ roleId, topicId });
+                      }
+                    });
+                  }
+                  // Add topic to junction with goal
+                  currentGoalId = cleanGoals.length + currentGoals.length - 1; // -1 because there is a dummy element at index 0
+                  cleanTopicGoals.push({ topicId, goalId: currentGoalId });
+                }
+              });
+            } else // it's either timeframe or status}
+            if (currentGoalName !== '') {
+              currentGoal[column] = el[key].trim();
+              const goalNum = goalColumn[1].slice(0, 1);
+              currentGoals[goalNum] = { ...currentGoals[goalNum], ...currentGoal };
+            }
+          } else { // we have a goal name
```

Comment:
Is there anyway to order the processing of each column into the order it appears in the CSV? so the first "Goal" column processed is the name, followed by the topic, timeframe and status. Would make it easier to follow what is going on.